skbio/sequence/_sequence.py

```diff
@@ -1141,6 +1164,92 @@ def __deepcopy__(self, memo):
         """
         return self._copy(True, memo)
 
+    def add(self, x, *intervals):
+        """ Adds a feature to the metadata object.
+
+        Parameters
+        ----------
+        x : skbio.sequence.feature
+            The feature object being added.
+        intervals : itteerable of intervals
+            A list of intervals associated with the feature
+
+        Note
+        ----
+        The intervals associated with a feature is assumed to be under
+        the `location` keyword by default.
+        """
+        if not isinstance(x, Feature):
+            raise ValueError('x is not an instance of `Feature`')
+
+        # TODO: The below will require careful consideration
+        # since this will be using a little more memory, since the whole
+        # feature object will be stored.
+        for interval in intervals:
+            loc = _polish_interval(interval)
+            if loc is not None:
+                start, end = loc
+                self.interval_metadata.add(start, end, x)
+
+        # TODO: The below will require careful consideration
+        # since this will be using a little more memory, since the whole
+        # feature object will be stored.
+        if x not in self.feature_metadata.keys():
+            self.feature_metadata[x] = []
+
+        self.feature_metadata[x] = list(map(_polish_interval, intervals))
+
+
+    def _query_interval(self, interval):
+        start, end = _polish_interval(interval)
+        features = self.interval_metadata.find(start, end)
+        return features
+
+    def _query_feature(self, key, value):
+        queries = []
+        for feature in self.feature_metadata.keys():
+            if feature[key] == value:
+                 queries.append(feature)
+        return queries
+
+    def query(self, *args, **kwargs):
```

Comment:
rename it to `query_interval_metadata`? kindof unwieldy. open for better name. it is not intuitive from the current name when users run `seq.query()`